### PR TITLE
Fixed table overflow for profits table. Added a scroller when horizontal view is less than 1800px

### LINKED ISSUE #24: BUG: Profits Table Overflow
## Summary
When the viewport is less than 1800px, the table on the user commons page overflows in it's container.

## Expected Behavior
The profits table should shrink with its container.

## Current / Observed Behavior
The table overflows on the common page.

## Steps to Reproduce
- Navigate any user commons page
- Shrink your viewport to less than 1800px

### PR DESCRIPTION
**Pull request by jgarcia-ucsb**
Fixes [#24](https://github.com/ucsb-cs156-s23/proj-happycows-s23-6pm-2/issues/24)

Modified the profits and profits table to prevent an overflow in the horizontal direction when the window is less than a certain size. Makes it easier for the user to read the table data. Added test-id's and an additional check in each test file in order to kill surviving mutants due to the div style. Also changed a stryker enable to a stryker restore, based on this post: [stryker slack](https://ucsb-cs156-s23.slack.com/archives/C050779M19P/p1685726928710669)

Before:
Screen Shot 2023-06-04 at 4 50 50 PM

After:
Screen Shot 2023-06-04 at 4 51 14 PM
Screen Shot 2023-06-04 at 4 51 28 PM
Screen Shot 2023-06-04 at 4 51 45 PM

### CODE REVIEW

File: frontend/src/main/components/Commons/Profits.js [hunk 1]
```diff
@@ -24,7 +24,9 @@ const Profits = ({ profits }) => {
                 <Card.Title>
                     You will earn profits from milking your cows everyday at 4am.
                 </Card.Title>
-                <ProfitsTable profits={profitsForTable} />
+                <div style={{ overflowX: "auto" }} data-testid="profits">
+                    <ProfitsTable profits={profitsForTable} />
+                </div>
```
Comment (chriscerie) [inline frontend/src/main/components/Commons/Profits.js]: This should just render `ProfitsTable` without a div container.

**Review by chriscerie — CHANGES_REQUESTED** [review]
Story: https://ucsb-cs156-s23.github.io/proj-happycows-s23-6pm-2/prs/38/storybook/?path=/story/components-commons-profits--uncontrolled

**Review by chriscerie — APPROVED** [review]
Thanks! This PR made me notice the 2 useMemo calls in `ProfitsTable.js` that are doing more harm than good both in terms of performance and complexity. It's not the optimization the author intended, but alas fixing it here is out of scope.

**Review by pconrad — APPROVED** [review]
LGTM

**Pull request merged**